torch/csrc/autograd/variable.h

```diff
@@ -286,7 +341,8 @@ struct TORCH_API Variable::Impl : public at::TensorImpl {
   explicit Impl(
       at::Tensor data,
       bool requires_grad = false,
-      Edge gradient_edge = Edge());
+      Edge gradient_edge = Edge(),
+      Variable::AutogradMeta* autograd_meta = new Variable::AutogradMeta());
```

Comment:
Woooooaah. This is pretty crazy.

1. Can we use a smart pointer here?
2. If you can't conveniently allocate a smart pointer in the default value (I didn't even know `new X` was legal here), can you just default it to nullptr and fill it in appropriately after the fact?